src/clj_irods/cache_tools.clj

```diff
@@ -85,9 +86,10 @@
   (let [cached (get-in @cache ks)]
     (if (and (delay? cached) (realized? cached))
       (delay (rethrow-if-error @cached))
-      (let [ag (agent nil)]
-        (send-via pool ag (fn [n] @(action)))
-        (delay (await ag) (rethrow-if-error @ag))))))
+      (otel/with-span [s ["agent for calculation"]]
+        (let [ag (agent nil)]
+          (send-via pool ag (fn [n] (with-open [_ (otel/span-scope s)] @(action))))
```

Comment:
This `with-open` ensures the opentelemetry scope goes into the stuff in the agent.